client/my-sites/site-settings/date-time-format/index.jsx

```diff
@@ -1,39 +1,178 @@
 /**
  * External dependencies
  */
-import React from 'react';
-import { connect } from 'react-redux';
+import React, { Component, PropTypes } from 'react';
 import { localize } from 'i18n-calypso';
-import page from 'page';
+import { capitalize, includes } from 'lodash';
 
 /**
  * Internal dependencies
  */
-import DateTimeFormatOptions from './date-time-format-options';
-import DocumentHead from 'components/data/document-head';
-import HeaderCake from 'components/header-cake';
-import { getSelectedSite } from 'state/ui/selectors';
-
-export const DateTimeFormat = ( { translate, site } ) => {
-	const goBack = () => {
-		page( '/settings/general/' + site.slug );
+import FoldableCard from 'components/foldable-card';
+import DateFormatOption from './date-format-option';
+import StartOfWeekOption from './start-of-week-option';
+import TimeFormatOption from './time-format-option';
+import { defaultDateFormats, defaultTimeFormats } from './default-formats';
+import {
+	getLocalizedDate,
+	phpToMomentDatetimeFormat,
+} from './utils';
+
+export class DateTimeFormat extends Component {
```

Comment:
If you were starting with this PR right now, I'd recommend splitting this PR to two:

* Changing Date & Time to inline and removing unused code and route.
* Moving the inline Date & time to Writing.

This would reduce the change of introducing subtle issues, and would allow the PR to be reviewed easier. But it's your call really.